### PR TITLE
[#4298]fix(flink-connector): fix scala version in flink connector

### PR DESCRIPTION
**Pull request by coolderli**
### What changes were proposed in this pull request?

Flink only supports Scala 2.12. All scala API will removed in the future version.

https://issues.apache.org/jira/browse/FLINK-23986
https://issues.apache.org/jira/browse/FLINK-20845
https://issues.apache.org/jira/browse/FLINK-13414

### Why are the changes needed?

Fix: #4298

### Does this PR introduce _any_ user-facing change?

- no

### How was this patch tested?

- `./gradlew flink-connector:publishToMavenLocal -PdefaultScalaVersion=2.13 `

image

### CODE REVIEW

**Comment by coolderli** [comment]
@FANNG1 Please take a look. Thanks.

**Comment by FANNG1** [comment]
we could exclude the Flink connector from Gravitino if the scala version is 2.13, like spark-connector in `setting.gradle.kts`
```
if (scalaVersion == "2.12") {
  include("spark-connector:spark-3.3", "spark-connector:spark-runtime-3.3")
  project(":spark-connector:spark-3.3").projectDir = file("spark-connector/v3.3/spark")
  project(":spark-connector:spark-runtime-3.3").projectDir = file("spark-connector/v3.3/spark-runtime")
}
```

File: flink-connector/build.gradle.kts [hunk 1]
```diff
@@ -27,7 +27,13 @@ repositories {
 }
 
 val flinkVersion: String = libs.versions.flink.get()
-val scalaVersion: String = project.properties["scalaVersion"] as? String ?: extra["defaultScalaVersion"].toString()
+
```
Comment (coolderli) [inline flink-connector/build.gradle.kts:30]: @FANNG1 I kept this change to avoid the `defaultScalaVersion` changing. What do you think?

Comment (FANNG1) [reply flink-connector/build.gradle.kts:30]: I'm ok

**Comment by FANNG1** [comment]
@yuqi1129 do you have time to review this PR?

**Comment by yuqi1129** [comment]
> @yuqi1129 do you have time to review this PR?

I have reviewed it and I'm okay with the change.

**Review by FANNG1 — APPROVED** [review]

**Pull request merged**